# Downoload data

In [2]:
import kagglehub

# Download latest version
path = kagglehub.dataset_download("yasserh/titanic-dataset", output_dir="./data")

print("Path to dataset files:", path)

100%|██████████| 22.0k/22.0k [00:00<00:00, 2.29MB/s]

Extracting files...
Path to dataset files: ./data


# 1. Load data

In [7]:
from pathlib import Path
import pandas as pd

DATA_DIR = Path("data")
DATA_FILE = DATA_DIR / "Titanic-Dataset.csv"

df = pd.read_csv(DATA_FILE)

df.head(7)

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S
5,6,0,3,"Moran, Mr. James",male,NaN,0,0,330877,8.4583,NaN,Q
6,7,0,1,"McCarthy, Mr. Timothy J",male,54.0,0,0,17463,51.8625,E46,S


# 2. Data inspection

## 2.1 Basic information about the data 

In [16]:
print(df.shape)
print(df.columns)

print()

df.info()

(891, 12)
Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='str')

<class 'pandas.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    str    
 4   Sex          891 non-null    str    
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    str    
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    str    
 11  Embarked     889 non-null    str    
dtypes: float64(2), int64(5), str(5)
memory usage: 83.7 KB


### Observation
The dataset contains **891 passengers and 12 columns**. The `Survived` column is our target variable, while the remaining columns contain passenger characteristics and information that may be useful for predicting survival.

We can also see that there are missing values in the `Age`, `Cabin`, and `Embarked` columns. These missing values will need to be addressed during the data preprocessing stage.

## 2.2 Statistics for numerical variables

In [18]:
df.describe()

,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


### Observation
From the numerical statistics, we can see that almost **62% of the passengers did not survive** the Titanic disaster.

The `Fare` variable is **right-skewed**, indicating that most passengers paid relatively low fares, while a small number of passengers paid significantly higher fares.

The statistics for `SibSp` and `Parch` suggest that most passengers **traveled without close family members** according to these variables.

More than **50% of the passengers were in the third class**, which indicates that third-class passengers represent the largest group in the dataset.


## 2.3 Categorical distributions

In [24]:
print(df["Sex"].value_counts())
print()
print(df["Embarked"].value_counts())
print()
print(df["Pclass"].value_counts())

Sex
male      577
female    314
Name: count, dtype: int64

Embarked
S    644
C    168
Q     77
Name: count, dtype: int64

Pclass
3    491
1    216
2    184
Name: count, dtype: int64


### Observation
The categorical distributions show that **men accounted for approximately two-thirds of the passengers, while women accounted for about one-third**.

Most passengers **embarked from Southampton**, making it the most common embarkation port. 

The **third class was also the largest passenger class** in the dataset.